Compile Warning - `DllExportedVariableReader32::GetInt`

When compiled on Linux with Clang, the following warning is produced:
```
DllExportedVariablesReader32.cpp:161:10: warning: cast to 'unsigned int *' from smaller integer type
      'std::uint32_t' (aka 'unsigned int') [-Wint-to-pointer-cast]
        return *(uint32_t*)GetExportedFileOffset(variableName);
```

The implementation of the function is:
```cpp
int DllExportedVariableReader32::GetInt(const std::string& variableName)
{
	return *(uint32_t*)GetExportedFileOffset(variableName);
}
```

Curiously the function right after is named similarly, but has a rather different implementation:
```cpp
bool DllExportedVariableReader32::GetBool(const std::string& variableName)
{
	stream.Seek(GetExportedFileOffset(variableName));
	
	bool value;
	stream.Read(value);

	return value;
}
```

As the `GetInt` method is not actually used in the project, I suspect the implementation does not behave as expected. I believe it should match the `stream.Read` code from the corresponding `GetBool` method.

----

Alternatively, the `GetStruct` implementation could actually be used for simple built-in data types, such as `uint32_t` and `bool`. Perhaps the template method should be renamed to indicate the more general nature it supports. The one restriction on use is the type should be trivially copyable. That would match the restriction on the `stream.Read` method in the OP2Utility project.

```cpp
template <typename StructType>
std::enable_if_t<std::is_trivially_copyable_v<StructType>, StructType>
ReadExport(const std::string& exportName) 
{
	stream.Seek(GetExportedFileOffset(exportName));

	StructType structValue;
	stream.Read(structValue);
	return structValue;
}
```

The `GetInt` and `GetBool` methods could then be removed.

--- a/DllExportedVariablesReader32.cpp
+++ b/DllExportedVariablesReader32.cpp
@@ -138,25 +138,25 @@ std::string DllExportedVariableReader32::ReadExportString(const std::string& exp
 	return ReadNullTerminatedString();
 }
 
-std::size_t DllExportedVariableReader32::GetExportOrdinal(const std::string& variableName)
+std::size_t DllExportedVariableReader32::GetExportOrdinal(const std::string& exportName)
 {
 	for (std::size_t i = 0; i < exportNameTable.size(); ++i) {
-		if (variableName == exportNameTable[i]) {
+		if (exportName == exportNameTable[i]) {
 			return i;
 		}
 	}
 
-	throw std::runtime_error("Requested exported variable name of " + variableName + " was not found in the DLL");
+	throw std::runtime_error("Requested exported variable name of " + exportName + " was not found in the DLL");
 }
 
-std::uint32_t DllExportedVariableReader32::GetExportedFileOffset(const std::string& variableName)
+std::uint32_t DllExportedVariableReader32::GetExportedFileOffset(const std::string& exportName)
 {
-	const auto rva = exportAddressTable[GetExportOrdinal(variableName)];
+	const auto rva = exportAddressTable[GetExportOrdinal(exportName)];
 
 	return RvaToFileOffset(rva, FindSectionTableContainingRva(rva));
 }
 
-bool DllExportedVariableReader32::DoesExportExist(const std::string& variableName)
+bool DllExportedVariableReader32::DoesExportExist(const std::string& exportName)
 {
-	return std::find(exportNameTable.begin(), exportNameTable.end(), variableName) != exportNameTable.end();
+	return std::find(exportNameTable.begin(), exportNameTable.end(), exportName) != exportNameTable.end();
 }

--- a/DllExportedVariablesReader32.cpp
+++ b/DllExportedVariablesReader32.cpp
@@ -132,9 +132,9 @@ bool DllExportedVariableReader32::IsDll(const CoffHeader& coffHeader)
 	return (coffHeader.characteristics & 0x2000) == 0x2000; // IMAGE_FILE_DLL
 }
 
-std::string DllExportedVariableReader32::GetString(const std::string& variableName)
+std::string DllExportedVariableReader32::ReadExportString(const std::string& exportName)
 {
-	stream.Seek(GetExportedFileOffset(variableName));
+	stream.Seek(GetExportedFileOffset(exportName));
 	return ReadNullTerminatedString();
 }
 

--- a/DllExportedVariablesReader32.cpp
+++ b/DllExportedVariablesReader32.cpp
@@ -156,21 +156,6 @@ std::uint32_t DllExportedVariableReader32::GetExportedFileOffset(const std::stri
 	return RvaToFileOffset(rva, FindSectionTableContainingRva(rva));
 }
 
-int DllExportedVariableReader32::GetInt(const std::string& variableName)
-{
-	return *(uint32_t*)GetExportedFileOffset(variableName);
-}
-
-bool DllExportedVariableReader32::GetBool(const std::string& variableName)
-{
-	stream.Seek(GetExportedFileOffset(variableName));
-	
-	bool value;
-	stream.Read(value);
-
-	return value;
-}
-
 bool DllExportedVariableReader32::DoesExportExist(const std::string& variableName)
 {
 	return std::find(exportNameTable.begin(), exportNameTable.end(), variableName) != exportNameTable.end();

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -19,7 +19,7 @@ public:
 	bool GetBool(const std::string& variableName);
 
 	template <typename DataType>
-	DataType GetStruct(const std::string& exportName)
+	DataType ReadExport(const std::string& exportName)
 	{
 		static_assert(std::is_trivially_copyable_v<DataType>, "Type must be trivially copyable");
 

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -5,6 +5,7 @@
 #include <vector>
 #include <string>
 #include <cstddef>
+#include <type_traits>
 
 
 // Access exported variables from a 32 bit DLL without loading the DLL into memory.
@@ -18,7 +19,8 @@ public:
 	bool GetBool(const std::string& variableName);
 
 	template <typename StructType>
-	StructType GetStruct(const std::string& exportName) 
+	std::enable_if_t<std::is_trivially_copyable_v<StructType>, StructType>
+	GetStruct(const std::string& exportName)
 	{
 		stream.Seek(GetExportedFileOffset(exportName));
 

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -15,8 +15,6 @@ public:
 	DllExportedVariableReader32(const std::string& filename);
 
 	std::string GetString(const std::string& variableName);
-	int GetInt(const std::string& variableName);
-	bool GetBool(const std::string& variableName);
 
 	template <typename DataType>
 	DataType ReadExport(const std::string& exportName)

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -19,9 +19,10 @@ public:
 	bool GetBool(const std::string& variableName);
 
 	template <typename StructType>
-	std::enable_if_t<std::is_trivially_copyable_v<StructType>, StructType>
-	GetStruct(const std::string& exportName)
+	StructType GetStruct(const std::string& exportName)
 	{
+		static_assert(std::is_trivially_copyable_v<StructType>, "Type must be trivially copyable");
+
 		stream.Seek(GetExportedFileOffset(exportName));
 
 		StructType structValue;

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -28,7 +28,7 @@ public:
 		return value;
 	}
 
-	bool DoesExportExist(const std::string& variableName);
+	bool DoesExportExist(const std::string& exportName);
 
 private:
 	Stream::FileReader stream;
@@ -43,6 +43,6 @@ private:
 	void LoadNameTable(std::uint32_t rva, const SectionTable& sectionTable, std::size_t count);
 	std::uint32_t RvaToFileOffset(std::uint32_t rva, const SectionTable& sectionTable);
 	std::string ReadNullTerminatedString(std::size_t maxCount = SIZE_MAX);
-	std::size_t GetExportOrdinal(const std::string& variableName);
-	std::uint32_t GetExportedFileOffset(const std::string& variableName);
+	std::size_t GetExportOrdinal(const std::string& exportName);
+	std::uint32_t GetExportedFileOffset(const std::string& exportName);
 };

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -18,16 +18,16 @@ public:
 	int GetInt(const std::string& variableName);
 	bool GetBool(const std::string& variableName);
 
-	template <typename StructType>
-	StructType GetStruct(const std::string& exportName)
+	template <typename DataType>
+	DataType GetStruct(const std::string& exportName)
 	{
-		static_assert(std::is_trivially_copyable_v<StructType>, "Type must be trivially copyable");
+		static_assert(std::is_trivially_copyable_v<DataType>, "Type must be trivially copyable");
 
 		stream.Seek(GetExportedFileOffset(exportName));
 
-		StructType structValue;
-		stream.Read(structValue);
-		return structValue;
+		DataType value;
+		stream.Read(value);
+		return value;
 	}
 
 	bool DoesExportExist(const std::string& variableName);

--- a/DllExportedVariablesReader32.h
+++ b/DllExportedVariablesReader32.h
@@ -14,7 +14,7 @@ class DllExportedVariableReader32
 public:
 	DllExportedVariableReader32(const std::string& filename);
 
-	std::string GetString(const std::string& variableName);
+	std::string ReadExportString(const std::string& exportName);
 
 	template <typename DataType>
 	DataType ReadExport(const std::string& exportName)

--- a/MissionTable.cpp
+++ b/MissionTable.cpp
@@ -68,9 +68,9 @@ void WriteRow(DllExportedVariableReader32& dllReader, std::string_view filename)
 	{
 		WriteCell(filename, columnWidths[0]);
 
-		WriteCell(dllReader.GetString("LevelDesc"), columnWidths[1]);
-		WriteCell(dllReader.GetString("MapName"), columnWidths[2]);
-		WriteCell(dllReader.GetString("TechtreeName"), columnWidths[3]);
+		WriteCell(dllReader.ReadExportString("LevelDesc"), columnWidths[1]);
+		WriteCell(dllReader.ReadExportString("MapName"), columnWidths[2]);
+		WriteCell(dllReader.ReadExportString("TechtreeName"), columnWidths[3]);
 
 		// Some missions do not store LevelDesc, MapName, and TechTreeName within AIModDesc
 		auto aiModDesc = dllReader.ReadExport<AIModDesc>("DescBlock");

--- a/MissionTable.cpp
+++ b/MissionTable.cpp
@@ -73,7 +73,7 @@ void WriteRow(DllExportedVariableReader32& dllReader, std::string_view filename)
 		WriteCell(dllReader.GetString("TechtreeName"), columnWidths[3]);
 
 		// Some missions do not store LevelDesc, MapName, and TechTreeName within AIModDesc
-		auto aiModDesc = dllReader.GetStruct<AIModDesc>("DescBlock");
+		auto aiModDesc = dllReader.ReadExport<AIModDesc>("DescBlock");
 		WriteCell(static_cast<MissionTypes>(aiModDesc.missionType), columnWidths[4]);
 		WriteCell(aiModDesc.numPlayers, columnWidths[5]);
 		WriteBoolCell(static_cast<bool>(aiModDesc.boolUnitMission), columnWidths[6]);

